drivers/can/can_mcan.h

```diff
@@ -8,14 +8,20 @@
 #ifndef ZEPHYR_DRIVERS_CAN_MCAN_H_
 #define ZEPHYR_DRIVERS_CAN_MCAN_H_
 
-#define NUM_STD_FILTER_ELEMENTS DT_PROP(DT_PATH(soc, can), std_filter_elements)
-#define NUM_EXT_FILTER_ELEMENTS DT_PROP(DT_PATH(soc, can), ext_filter_elements)
-#define NUM_RX_FIFO0_ELEMENTS   DT_PROP(DT_PATH(soc, can), rx_fifo0_elements)
-#define NUM_RX_FIFO1_ELEMENTS   DT_PROP(DT_PATH(soc, can), rx_fifo0_elements)
-#define NUM_RX_BUF_ELEMENTS     DT_PROP(DT_PATH(soc, can), rx_buffer_elements)
+#ifdef CONFIG_CAN_MCUX_MCAN
+#define MCAN_DT_PATH DT_NODELABEL(can0)
```

Comment:
Can we do something like

```C
#define MCAN_DT_PATH DT_COMPAT_GET_ANY_STATUS_OKAY
```
?

Reply:
The current approach does not allow for different devicetree configurations of the M_CAN message RAM for multiple M_CAN frontend instances, so it needs to be reworked.

I'd prefer to leave this for the general refactoring of the M_CAN driver once the other frontend drivers are in (#41549 and #41674).